### ISSUE TITLE
关于输入数据的问题

### ISSUE THREAD

**Issue by 1661021512** [Fediory/HVI-CIDNet#21]
感谢作者的工作。我在使用您的代码时，当输入数据的shape为[1, 3, 426, 640]时，出现了如下问题：
![屏幕截图 2024-07-12 161607](https://github.com/user-attachments/assets/bf1d697d-6ebc-4e89-a981-18cdc56d075f)
这个错误来自于HVI_CIDNet\net\transformer_utils.py的第65行的x = torch.cat([x, y],dim=1)语句。
请问输入CIDNet的图片的尺寸是否有特殊要求？

**Comment by Fediory** [comment]
您好，因为我们的UNet有三次下采样，每次都是将tensor的Height和Width除以二，因此H与W需要被8整除。如果无法被整除的话，需要对输入图像进行padding预处理。您可以参考下面的代码(`./data/eval_sets.py`)：
` factor = 8
            h, w = input.shape[1], input.shape[2]
            H, W = ((h + factor) // factor) * factor, ((w + factor) // factor) * factor
            padh = H - h if h % factor != 0 else 0
            padw = W - w if w % factor != 0 else 0
            input = F.pad(input.unsqueeze(0), (0,padw,0,padh), 'reflect').squeeze(0)`

**Comment by 1661021512** [comment]
感谢您的回复，问题现在解决了
